From af7ca47c872cbe6a086a1fbdd0b4403b2ebd329c Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 25 Aug 2010 10:22:40 +1000
Subject: [PATCH 2/9] xsetwacom: support aboveof, belowof, leftof, rightof in xsetwacom TwinView.

I don't actually know what this is really supposed to do in the end, the
driver does the same thing irrespective of whether leftof or rightof is
specified.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Acked-by: Przemo Firszt <przemo@firszt.eu>
Acked-by: Ping Cheng <pinglinux@gmail.com>

Conflicts:

	tools/xsetwacom.c

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 tools/xsetwacom.c |   16 ++++++++++++++--
 1 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/tools/xsetwacom.c b/tools/xsetwacom.c
index aa4e757..708fa68 100644
--- a/tools/xsetwacom.c
+++ b/tools/xsetwacom.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2009 Red Hat, Inc.
+ * Copyright 2009 - 2010 Red Hat, Inc.
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License
@@ -1559,6 +1559,14 @@ static void set_twinview(Display *dpy, XDevice *dev, param_t* param, int argc, c
 		twinview = TV_LEFT_RIGHT;
 	else if (strcasecmp(argv[0], "vertical") == 0)
 		twinview = TV_ABOVE_BELOW;
+	else if (strcasecmp(argv[0], "belowof") == 0)
+		twinview = TV_ABOVE_BELOW;
+	else if (strcasecmp(argv[0], "aboveof") == 0)
+		twinview = TV_BELOW_ABOVE;
+	else if (strcasecmp(argv[0], "rightof") == 0)
+		twinview = TV_LEFT_RIGHT;
+	else if (strcasecmp(argv[0], "leftof") == 0)
+		twinview = TV_RIGHT_LEFT;
 	else
 		goto error;
 
@@ -1588,7 +1596,9 @@ static void set_twinview(Display *dpy, XDevice *dev, param_t* param, int argc, c
 	return;
 
 error:
-	fprintf(stderr, "Usage: xsetwacom rotate <device name> [NONE | CW | CCW | HALF]\n");
+	fprintf(stderr, "Usage: xsetwacom <device name> TwinView [<value>]\n");
+	fprintf(stderr, "Values: none, horizontal, vertical, belowof,"
+			"aboveof, leftof, rightof\n");
 	return;
 }
 
@@ -1824,6 +1834,8 @@ static void get_twinview(Display *dpy, XDevice *dev, param_t *param, int argc, c
 		case TV_NONE: twinview = "none"; break;
 		case TV_ABOVE_BELOW: twinview = "vertical"; break;
 		case TV_LEFT_RIGHT: twinview = "horizontal"; break;
+		case TV_RIGHT_LEFT: twinview = "leftof"; break;
+		case TV_BELOW_ABOVE: twinview = "aboveof"; break;
 		default:
 				    break;
 	}
-- 
1.7.2.2

